internal/server/singleprocess/service_runner.go

```diff
@@ -67,6 +69,156 @@ func (s *service) RunnerGetDeploymentConfig(
 	}, nil
 }
 
+func (s *service) AdoptRunner(
+	ctx context.Context,
+	req *pb.AdoptRunnerRequest,
```

Comment:
It'd be good to get these service functions in the runner with the `serverptype` Request validations like our other service funcs have.

Reply:
Done!